core/src/executor/delete.rs

```diff
@@ -14,27 +14,46 @@ use {
     thiserror::Error as ThisError,
 };
 
+use crate::prelude::Key;
+use futures::Future;
+use std::pin::Pin;
+
 #[derive(ThisError, Serialize, Debug, PartialEq, Eq)]
 pub enum DeleteError {
-    #[error("referencing column exists: {0}")]
-    ReferencingColumnExists(String),
+    #[error("Restrict reference column exists: {0}")]
+    RestrictingColumnExists(String),
 
```

Comment:
What was the purpose of this name `RestrictingColumnExists`?
When I decided name `ReferencingColumnExists`, I focused on `Referencing` cuz FK has two directions `Referencing` and `Referenced` by which both of Rust developer and DB User can be confused.

Reply:
That makes sense - I think I just meant to write `ReferencingColumnExists` and then I paused for a coffee and my brain reset. I will fix that shortly.

Reply:
Ah yes, now looking at the code I recalled - The thing is that error is only raised when a Restricting constraint is in place, it is not about solely a Referencing column. This way, the error was more precise, detailing what was causing it.

Reply:
Oh.. What i'm worried about is this part.
```rs
                        NoAction => {
                            return Err(DeleteError::RestrictingColumnExists(format!(
                                "{referencing_table_name}.{referencing_column_name}"
                            ))
                            .into());
                        }
```

https://github.com/gluesql/gluesql/pull/1517/files#diff-fb5db5309bc066a3ff406c630d67a7e4f0d9fcde46ab477e920a197a6f7c6cb4R106-R110

Cuz the actual arguments of Error are originated from `referencing~`.
And This is the only error case if there is Referencing column among 4 `ReferentialAction`s so that i thought `ReferencingColumnExists` would be enough.
And now even `ReferentialAction::Restrict` disappeared..

Reply:
I am not sure I understood your comment - which part would you like me to change?

Also, Restrict was removed as in another note you stated it should be handled as NoAction, so now I removed that variant. Could you clarify what is the problem there?

Reply:
I want to keep the name `ReferencingColumnExists`

Reply:
Ok, reverting change.

Reply:
Reverted in commit [56e7e9b](https://github.com/gluesql/gluesql/pull/1517/commits/56e7e9bb5a6bbc3d505ffc3ee357c188e9e736a5)